### PR TITLE
Feat: Core SDK improvements necessary for token gated mini apps

### PR DESCRIPTION
**Pull request by davidfurlong**
This is https://github.com/mod-protocol/mod/pull/92 without the Lit protocol mini apps.

- Adds a `combobox` element
- Adds a `select` element
- Adds a `textarea` element
- Adds a draft for `Model Definitions` published by Mini-apps that can be expressed in `schema.org` schemas
- Adds a `loadingLabel` prop to buttons
- Adds an `ethPersonalSign` mod `Action`
- Adds a new schema for metadata in `json-ld`
- Adds `json-ld` indexing to the metadata indexer (arweave and local)
- Adds a new endpoint for querying the metadata indexer by url, `/api/cast-embeds-metadata/by-url`

Open todos: 
- [x] needs changesets
- [x] Documentation for Model definitions
- [x] Consider adding `json-ld` to the metascraper

Would particularly like a review on this proposed way of doing Model definitions. Is there a strong case that these should be separate from Mini-apps entirely instead?

### CODE REVIEW

File: docs/pages/metadata-cache.mdx [hunk 1]
```diff
@@ -73,8 +75,75 @@ Returned metadata objects conform to the `UrlMetadata` type. This can then be us
...
+    // ...
+  },
+  "https://twitter.com": {
+    "title": "Example Title",
+    "description": "Example Description",
+    "image": "https://example.com/image.png"
+    // ...
+  }
+}
+```
```
Comment (stephancill) [inline docs/pages/metadata-cache.mdx:140]: `image` key is actually an object. 

see actual response:

```
{
  "https://google.com": {
    "image": {
      "url": "https://www.google.com/images/branding/googleg/1x/googleg_standard_color_128dp.png",
      "height": 128,
      "width": 128
    },
    "description": "Celebrate Native American Heritage Month with Google",
    "title": "Google",
    "publisher": "google.com",
    "logo": {
      "url": "https://www.google.com/favicon.ico"
    },
    "mimeType": "text/html; charset=UTF-8"
  },
  "https://twitter.com": {
    "description": "From breaking news and entertainment to sports and politics, get the full story with all the live commentary.",
    "title": "@ on Twitter",
    "publisher": "Twitter",
    "logo": {
      "url": "https://abs.twimg.com/favicons/twitter.3.ico"
    },
    "mimeType": "text/html; charset=utf-8"
  }
}
```

File: examples/api/next.config.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 module.exports = {
   reactStrictMode: true,
-  transpilePackages: ["@mod-protocol/react"],
+  transpilePackages: [
+    "@mod-protocol/react",
+    // Fixes https://discord.com/channels/896185694857343026/1174716239508156496
+    "@lit-protocol/bls-sdk",
+  ],
```
Comment (stephancill) [inline examples/api/next.config.js:7]: Should this change + lit-related packages in package.json not be in the token gated casts PR?

File: examples/api/package.json [hunk 3]
```diff
@@ -10,16 +10,24 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@irys/sdk": "^0.0.4",
+    "@lit-protocol/lit-node-client": "^3.0.24",
+    "@lit-protocol/types": "^2.2.61",
     "@mod-protocol/core": "^0.0.2",
     "@reservoir0x/reservoir-sdk": "^1.8.4",
     "@vercel/postgres-kysely": "^0.6.0",
+    "bip39": "^3.1.0",
```
Comment (stephancill) [inline examples/api/package.json:19]: `bip39` is unused as far as i can tell

File: packages/core/src/renderer.ts [hunk 4]
```diff
@@ -390,7 +447,37 @@ export class Renderer {
...
+    //       },
+    //       chainId: "7777777",
+    //       explorer: {
+    //         name: "Explorer",
+    //         url: "https://explorer.zora.energy",
+    //       },
+    //     },
+    //   },
+    // },
+  };
```
Comment (stephancill) [inline packages/core/src/renderer.ts:480]: Can probably remove, was used for debugging

File: packages/core/src/renderer.ts [hunk 5]
```diff
@@ -779,6 +881,73 @@ export class Renderer {
...
+          setTimeout(() => {
+            this.onEthPersonalSignAction(
+              {
+                data: {
+                  // domain: this.replaceInlineContext(action.data.domain),
+                  // address: this.replaceInlineContext(action.data.address),
+                  statement: this.replaceInlineContext(action.data.statement),
+                  // uri: this.replaceInlineContext(action.data.uri),
+                  version: this.replaceInlineContext(action.data.version),
+                  chainId: this.replaceInlineContext(action.data.chainId),
```
Comment (stephancill) [inline packages/core/src/renderer.ts:895]: Should `domain`, `address`, `uri` not be exposed?

File: packages/miniapp-registry/src/index.ts [hunk 6]
```diff
@@ -9,12 +9,14 @@ import NFTMinter from "@miniapps/nft-minter";
...
 import ImageRender from "@miniapps/image-render";
 import ChatGPTShorten from "@miniapps/chatgpt-shorten";
+import ZoraNftMinter from "@miniapps/zora-nft-minter";
 
 export const allMiniApps = [
   InfuraIPFSUpload,
   LivepeerVideo,
   GiphyPicker,
   VideoRender,
+  ZoraNftMinter,
```
Comment (stephancill) [inline packages/miniapp-registry/src/index.ts:19]: Shouldn't be included yet

**Review by stephancill — CHANGES_REQUESTED** [review]
lgtm, added some minor comments

File: packages/core/src/renderer.ts (hunk 5 above)
Comment (davidfurlong) [reply packages/core/src/renderer.ts:895]: this is done by the implementing client in the action handler

Comment (stephancill) [reply packages/core/src/renderer.ts:865]: ah okay makes sense

**Review by stephancill — APPROVED** [review]
`ethers` in `api/package.json` is unused. 

Otherwise lgtm

**Pull request merged**